tests/v2/validation/nodescaling/scaling_custom_cluster_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build (validation || infra.rke2k3s || cluster.custom || stress) && !infra.any && !infra.aks && !infra.eks && !infra.gke && !infra.rke1 && !cluster.any && !cluster.nodedriver && !sanity && !extended
+
+package nodescaling
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/scalinginput"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type CustomClusterNodeScalingTestSuite struct {
+	suite.Suite
+	client        *rancher.Client
+	session       *session.Session
+	scalingConfig *scalinginput.Config
+}
+
+func (s *CustomClusterNodeScalingTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *CustomClusterNodeScalingTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.scalingConfig = new(scalinginput.Config)
+	config.LoadConfig(scalinginput.ConfigurationFileKey, s.scalingConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *CustomClusterNodeScalingTestSuite) TestScalingCustomClusterNodes() {
+	nodeRolesEtcd := machinepools.NodeRoles{
+		Etcd:     true,
+		Quantity: 1,
+	}
+
+	nodeRolesControlPlane := machinepools.NodeRoles{
+		ControlPlane: true,
+		Quantity:     1,
+	}
+
+	nodeRolesWorker := machinepools.NodeRoles{
+		Worker:   true,
+		Quantity: 1,
+	}
+
```

Comment:
I've got the same question for this one, please ignore it if it's only 2.7 test case

Reply:
Same as above.

Reply:
Addressed